### PR TITLE
Refactor tiger binaries into `tiger-bin-shared` 

### PR DESCRIPTION
**Pull request by my4ng**
This also separates the self-update functionality into the aforementioned package.

### CODE REVIEW

File: ck3-tiger/Cargo.toml [hunk 1]
```diff
@@ -15,17 +15,15 @@ rust-version = "1.70"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-tiger-lib = { version = "0.9.5", path = "..", features = ["ck3"] }
+tiger-lib = { path = "..", features = ["ck3"] }
```
Comment (amtep) [inline ck3-tiger/Cargo.toml:18]: Hmm you removed the version. Will this still pull in the latest tiger-lib when updating ck3-tiger?

File: tiger-bin-shared/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ck3 = ["tiger-lib/ck3"]
+vic3 = ["tiger-lib/vic3"]
+imperator = ["tiger-lib/imperator"]
+
+[target.'cfg(target_os = "windows")'.dependencies]
+self_update = { version = "0.39", features = [
+    "archive-zip",
+    "compression-zip-deflate",
+]}
+winreg = "0.52"
```
Comment (amtep) [inline tiger-bin-shared/Cargo.toml:34]: For some reason the CI check complains about lack of winreg

File: tiger-bin-shared/src/binary.rs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    /// installed over the current binary.
+    Update {
+        /// release version (e.g. 0.9.3)
+        version: Option<String>,
+    },
+}
+
+#[derive(Args)]
+struct ValidateArgs {
+    /// Path to .mod file of mod to check.
```
Comment (amtep) [inline tiger-bin-shared/src/binary.rs:56]: This is awkward because Vic3 has no .mod files

**Review by amtep — CHANGES_REQUESTED** [review]
Can you move the gamedir module as well?

And can you do the _game_-tiger-auto binaries? Imperator doesn't currently have one but it should, for symmetry.

File: ck3-tiger/Cargo.toml (hunk 1 above)
Comment (my4ng) [reply ck3-tiger/Cargo.toml:18]: yes you are right won't be able to publish on crates.io otherwise. will fix

**Review by amtep — DISMISSED** [review]

**Review by amtep — APPROVED** [review]

**Pull request merged**